src/sonic-bgpcfgd/bgpcfgd/managers_advertise_rt.py

```diff
@@ -0,0 +1,108 @@
+from .manager import Manager
+from .template import TemplateFabric
+from swsscommon import swsscommon
+
+
+class AdvertiseRouteMgr(Manager):
+    """ This class Advertises routes when ADVERTISE_NETWORK table in STATE_DB is updated """
+    def __init__(self, common_objs, db, table):
+        """
+        Initialize the object
+        :param common_objs: common object dictionary
+        :param db: name of the db
+        :param table: name of the table in the db
+        """
+        super(AdvertiseRouteMgr, self).__init__(
+            common_objs,
+            [],
+            db,
+            table,
+        )
+        
+        self.directory.subscribe([("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"),], self.on_bgp_asn_change)
+        self.advertised_routes = dict()
+
+
+    OP_DELETE = 'DELETE'
+    OP_ADD = 'ADD'
+
+
+    def set_handler(self, key, data):
```

Comment:
does the set handler invoked when bgp is restarted? please confirm

Reply:
I checked the configuration after bgp restart and it would be invoked after bgp restart.